### ISSUE TITLE
`nodeVersion: false` doesn't disable the related rules

### ISSUE THREAD

**Issue by XhmikosR** [xojs/xo#598]
```
C:\Users\xmr\Desktop\foo>node -v && npm -v && ver
v14.17.5
6.14.14
Microsoft Windows [Version 10.0.19043.1165]

C:\Users\xmr\Desktop\foo>npm ls xo
foo@1.0.0 C:\Users\xmr\Desktop\foo
`-- xo@0.44.0
```

I was trying to disable the following two rules:

* "node/no-unsupported-features/es-builtins"
* "node/no-unsupported-features/es-syntax"

but it doesn't have any effect. So then I saw that there's the `nodeVersion` option that could control this. I tried setting it to `false` but it doesn't work either.

A quick patch would be to change https://github.com/xojs/xo/blob/bc6e05de5ad89e2e9d72070c7d4a1b04c0f4bd12/lib/options-manager.js#L205

```diff
-		...(enginesOptions && enginesOptions.node && semver.validRange(enginesOptions.node) ? {nodeVersion: enginesOptions.node} : {}),
+		...(xoOptions.nodeVersion !== false && enginesOptions && enginesOptions.node && semver.validRange(enginesOptions.node) ? {nodeVersion: enginesOptions.node} : {}),
```

But shouldn't it be possible to override the related rules?

**Comment by sindresorhus** [comment]
> A quick patch would be to change

Pull request welcome.

**Comment by XhmikosR** [comment]
I can make a PR but I'm still not sure if it's the proper solution since IMHO it should be possible to override those rules only without overriding `nodeVersion`.

**Comment by sindresorhus** [comment]
> But shouldn't it be possible to override the related rules?

Ideally, yes, but it's complicated. Rule overrides can come from your package.json, but it can also come from a shareable config, and we override before all the config is resolved, so we cannot reliably check whether it's already set.

https://github.com/xojs/xo/blob/bc6e05de5ad89e2e9d72070c7d4a1b04c0f4bd12/lib/options-manager.js#L309-L313

**Comment by fregante** [comment]
It looks like this was not fixed in #761, `"nodeVersion": false,` won't disable `n` rules


```
source/github-events/on-react-page-update.tsx:8:25
  ✖    8:25  The AbortSignal.any is still an experimental feature and is not supported until Node.js 20.3.0 (backported: ^18.17.0). The configured version range is >= 20.  n/no-unsupported-features/node-builtins
```

It did fix https://github.com/xojs/xo/issues/613 though.

Reopen?

**Comment by fregante** [comment]
Pretty sure `nodeVersion: false` is being overridden here:

https://github.com/xojs/xo/blob/b537b21355cbeb6f816c5fdee6b45abf3f593750/lib/options-manager.js#L120

because `xoOptions.nodeVersion` is `false` but `options.nodeVersion` is still set to what's in the package.json after this line, so that's what's received by the rest of the code

**Comment by fregante** [comment]
Confirmed to be fixed in v0.59.2

- https://github.com/refined-github/refined-github/pull/7588
